SALTO-1650 Support OAuth in Zendesk (#2501)
salto-io/salto: packages/zendesk-support-adapter/README.md

We do this using the [Zendesk REST API](<https://developer.zendesk.com/api-reference/>).

## Configuring your Zendesk account for Salto OAuth Authentication
Salto supports authenticating with Zendesk using either a combination of user-name and password, or using OAuth authentication. In order to use OAuth authentication, please follow the steps below.

> - Create a new "OAuth Client" in your Zendesk account (see https://support.zendesk.com/hc/en-us/articles/4408845965210#topic_s21_lfs_qk)

Reviewer: maybe link to https://developer.zendesk.com/documentation/ticketing/working-with-oauth/creating-and-using-oauth-tokens-with-the-api/#create-an-oauth-client instead? also styling nit:
```suggestion
- Create a new "OAuth Client" in your Zendesk account (see [details](https://developer.zendesk.com/documentation/ticketing/working-with-oauth/creating-and-using-oauth-tokens-with-the-api/#create-an-oauth-client))
```

> 	- Under "Redirect URL", write <http://localhost:PORT>, where PORT is a port you can allow salto to open momentarily on your computer (it will not be open to external connections). Except for PORT, do not change any character in the URL.
> - You can now connect your Zendesk account using OAuth credentials, by adding the flag '-a oauth' or '--auth-type oauth' to the commands 'salto service add zendesk_support' or 'salto service login zendesk_support'

Reviewer: nits
```suggestion
	- Under "Redirect URL", write `http://localhost:PORT`, where `PORT` is a port you can allow salto to open momentarily on your computer (it will not be open to external connections). Except for PORT, do not change any character in the URL.
- You can now connect your Zendesk account using OAuth credentials, by adding the flag `-a oauth` or `--auth-type oauth` to the commands `salto service add zendesk_support` or `salto service login zendesk_support`
```

> - You will be asked to provide the Client ID of the OAuth Client, the generated Client Secret, the port you provided in the redirect URL, and your Zendesk account subdomain.

Reviewer: nit: since these are a lot of parameters, maybe split them into bullets?
